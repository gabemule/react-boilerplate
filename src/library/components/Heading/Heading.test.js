import React from 'react'
import { render } from 'enzyme'

// components
import { DsHeading } from './index'

// config
import HeadingConfig from './Heading.config'

// constants
const HeadingClass = HeadingConfig.class
const HeadingPropText = 'Heading Default Text'

describe('Heading Component', () => {
  it('Matches Snapshot', () => {
    const wrapper = render(
      <DsHeading type={'display'}>{HeadingPropText}</DsHeading>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    const wrapper = render(
      <DsHeading type={'display'}>{HeadingPropText}</DsHeading>
    )
    expect(wrapper).toBeTruthy()
    expect(wrapper.text()).toBe(HeadingPropText)
    expect(wrapper.find(HeadingClass)).toBeTruthy()
  })
})

describe('Props', () => {
  HeadingConfig.props.type.options.forEach((type) => {
    it(`Accept prop type: ${type}`, () => {
      const wrapper = render(
        <DsHeading type={type}>{HeadingPropText}</DsHeading>
      )
      expect(wrapper.find(`${HeadingClass}--${type}`)).toBeTruthy()
    })
  })
})
