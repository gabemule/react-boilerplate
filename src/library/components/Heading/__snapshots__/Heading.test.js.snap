// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component Matches Snapshot 1`] = `
LoadedCheerio {
  "0": <h1
    class="ds-heading ds-heading--display"
    data-testid="text-heading"
  >
    Heading Default Text
  </h1>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
